### DISCUSSION TITLE
[Feature Request] Can 86Box emulate support for parallel port ZIP/CD-ROMs and tape drives?

### DISCUSSION THREAD

**Discussion by ppgrainbow** [86Box/86Box#4051] [Ideas]
I don't know if it's possible to emulate support for parallel port versions of Iomega ZIP, CD-ROM and even external hard drives on any of the three LPT ports?

Also, would it also be possible to implement support for parallel port as well as IDE or SCSI tape drives?

This is something that Windows NT 3.x and 4.0 needs, because under these OSes, it's not even possible to back up data onto a floppy drive or ZIP drive without the need for 3rd party software.

Thank you for your time. :)

**Comment by OBattler** [comment]
This is going to require to first emulate ECP and EPP parallel ports, which is in the plans for 86Box v2.20.

**Comment by waltje** [comment]
These work on standard ports, too, just in slow 4-bit mode.

Those interfaces basically implement an NCR5380 on the "remote" parport "bus".

--waltje


     On Thursday, June 18, 2020, 04:41:13 PM EDT, OBattler <notifications@github.com> wrote:  
 
 


This is going to require to first emulate ECP and EPP parallel ports, which is in the plans for 86Box v2.20.

**Comment by ppgrainbow** [comment]
> 
> 
> This is going to require to first emulate ECP and EPP parallel ports, which is in the plans for 86Box v2.20.

Thank you for the input. How can we implement and emulate the ECP and EPP parallel ports in 86Box v2.20?

**Comment by thunderbird32** [comment]
Emulation of tape drives would extremely helpful, particularly SCSI-based QIC drives. Many early releases of UNIX for the PC were released on QIC tape (Interactive UNIX, Dell UNIX, SCO XENIX, etc), and in some cases those are the only extant versions.

**Comment by fborges22** [comment]
Recently, I have submitted this PR to support tape drives and backup/restore.

https://github.com/86Box/86Box/pull/7792
